Catalog/Catalog.cpp

```diff
@@ -2765,16 +2768,21 @@ void Catalog::createShardedTable(
     const list<ColumnDescriptor>& cols,
     const std::vector<Parser::SharedDictionaryDef>& shared_dict_defs) {
   cat_write_lock write_lock(this);
+  DataframeTableDescriptor* df_td = dynamic_cast<DataframeTableDescriptor*>(&td);
```

Comment:
This doesn't seem quite right -- can you explain?

Reply:
This's added for determining which of type of descriptor is using now. In case when `td` contains `TableDescriptor` object `df_dt` will be `nullptr` and we initialize `df_dt` with DataframeTableDescriptor(TableDescriptor) constructor.  It's needed as we create locally objects of descriptors at lines 2778 and 2785. In general, this dynamic cast may be moved to `DataframeTableDescriptor` constructor